### ISSUE TITLE
exec error: oci runtime error: exec failed: exit status 1

### ISSUE THREAD

**Issue by npcode** [docker/docker#22494]
**Output of `docker version`:**

```
Client:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   20f81dd
 Built:        Thu Mar 10 15:39:25 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.0-dev
 API version:  1.24
 Go version:   go1.5.4
 Git commit:   8a0d2d8
 Built:        Sat Apr 23 10:24:19 2016
 OS/Arch:      linux/amd64
 Experimental: true
```

**Output of `docker info`:**

```
Containers: 42
 Running: 9
 Paused: 0
 Stopped: 33
Images: 32
Server Version: 1.12.0-dev
Storage Driver: overlay
 Backing Filesystem: xfs
Logging Driver: json-file
Plugins: 
 Volume: local
 Network: bridge null host
Kernel Version: 4.5.0-1.el7.elrepo.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 24
Total Memory: 47.13 GiB
Name: host.name
ID: EAQU:N4MD:AIFV:TRN5:UWXR:6XWR:M5J3:NN2P:YLAV:X4QT:JS6M:BAYI
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
Experimental: true
Cluster store: consul://11.11.111.191:8500/aaa
Cluster advertise: 11.11.111.13:2376
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

physical

**Steps to reproduce the issue:**
1. Run `docker exec foo -it /bin/bash`

**Describe the results you received:**

```
rpc error: code = 2 desc = "oci runtime error: exec failed: exit status 1: time=\"2016-05-04T13:50:24+09:00\" level=fatal msg=\"dial unix /var/lib/docker/network/files/58f8a9c30737db8d432c22f770a15e70bc85fe1d8a5c148c6578709975620d50.sock: connect: connection refused\" \n"% 
```

**Describe the results you expected:**

bash shell

**Additional information you deem important (e.g. issue happens only occasionally):**

**Comment by femueller** [comment]
+1

**Comment by thaJeztah** [comment]
I notice the command you're using is incorrect, because the `-it` is put after the container name, I suspect

    docker exec foo -it /bin/bash

Should be

    docker exec -it foo /bin/bash

Can you provide some more information;

- Can you also show the steps to start the container you're trying to exec in? Is that container running?
- I see that a cluster store is mentioned in your configuration; are you using overlay networking? Is this node part of a Swarm cluster?
- You're running on kernel 4.5. CentOS 7 is shipping with a 3.10 kernel; did you install a custom kernel?

**Comment by npcode** [comment]
> Can you also show the steps to start the container you're trying to exec in? Is that container running?

Yes, the container was running.

> I see that a cluster store is mentioned in your configuration; are you using overlay networking? Is this node part of a Swarm cluster?

Yes, I am using overlay network and the docker daemon is also a node of a swarm cluster.

> You're running on kernel 4.5. CentOS 7 is shipping with a 3.10 kernel; did you install a custom kernel?

Yes, I did.

After I posted this issue I restarted the container then it produces much better error message as follows:

```
$ docker exec foo -it /bin/bash                                                                                                                                                                                        16:32:26
rpc error: code = 2 desc = "oci runtime error: exec failed: exec: \"-it\": executable file not found in $PATH"%  
```

**Comment by thaJeztah** [comment]
@npcode that error is expected, because you put `-it` *after* the container name, not before, so docker is trying to run `-it` as the container command. Try doing `docker run -it foo bash`

**Comment by npcode** [comment]
@thaJeztah Thanks. That error clearly showed my mistake.

**Comment by thaJeztah** [comment]
Does that resolve your issue?

**Comment by npcode** [comment]
@thaJeztah Yes, the new error is clear enough and I can't reproduce the first confusing "connection refused" error.

**Comment by thaJeztah** [comment]
Alright, let me close this issue then, thanks!

**Comment by 1e1** [comment]
Sorry, I have the same issue: 
docker exec -ti test_cassandra_1 bash
`rpc error: code = 2 desc = "oci runtime error: exec failed: exit status 1"

$docker version
```
Client:
 Version:      1.11.0
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   4dc5990
 Built:        Wed Apr 13 18:34:23 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.0
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   4dc5990
 Built:        Wed Apr 13 18:34:23 2016
 OS/Arch:      linux/amd64
```

docker info
```
Containers: 41
 Running: 5
 Paused: 0
 Stopped: 36
Images: 27
Server Version: 1.11.0
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 387
 Dirperm1 Supported: false
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: null host bridge
Kernel Version: 3.13.0-77-generic
Operating System: Ubuntu 14.04.4 LTS
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 3.861 GiB
Name: dockercompose-ubuntu-4gb-ams3-01
ID: ZX6M:WSMT:MQTI:CC4T:N5FD:JQ2K:OW7A:66FD:KELP:FEV2:HWMT:73LX
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): false
Registry: https://index.docker.io/v1/
```

Environnement: Digital Ocean (but it works locally)

Steps to reproduce: 
`docker exec -ti test_cassandra_1 bash` or `docker exec -ti test_cassandra_1 /bin/bash`

**Comment by thaJeztah** [comment]
@1e1 is the `test_cassandra_1` container running?

**Comment by 1e1** [comment]
Yes
I start by `docker-compose up`, then I "break" the command line by `^C`. My containers stop except `test_cassandra_1`

**Comment by wicope** [comment]
Same, only once for now

```
vagrant@prod:~$ docker version
Client:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:47:50 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:47:50 2016
 OS/Arch:      linux/amd64
```

```
vagrant@prod:~$ docker info
Containers: 4
 Running: 4
 Paused: 0
 Stopped: 0
Images: 7
Server Version: 1.11.2
Storage Driver: devicemapper
 Pool Name: docker-8:1-262184-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: ext4
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 2.29 GB
 Data Space Total: 107.4 GB
 Data Space Available: 34.63 GB
 Metadata Space Used: 3.441 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.144 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Either use `--storage-opt dm.thinpooldev` or use `--storage-opt dm.no_warn_on_loop_devices=true` to suppress this warning.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.77 (2012-10-15)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: null host bridge
Kernel Version: 3.13.0-86-generic
Operating System: Ubuntu 14.04.4 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 489.6 MiB
Name: prod
ID: BNTB:UM4F:RHXI:MI6Y:FLJP:XYRD:PU5S:7TLQ:6LFA:5QV7:KQA2:PDF2
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
```

```
docker exec -it 6f4bf6ce5b5d /bin/bash
rpc error: code = 2 desc = "oci runtime error: exec failed: exec: \"/bin/bash\": stat /bin/bash: no such file or directory
```

For this case i use:
`docker exec -it foo ash`

**Comment by ecejas** [comment]
its -it not -ti:
docker exec -it test_cassandra_1 /bin/bash
